programs/axelar-solana-its/src/processor/interchain_token.rs

```diff
@@ -590,6 +592,13 @@ pub(crate) fn approve_deploy_remote_interchain_token(
     )?;
 
     let token_id = crate::interchain_token_id(&deployer, &salt);
+    let (its_root_pda, _) = find_its_root_pda();
+    let (token_manager_check, _) = find_token_manager_pda(&its_root_pda, &token_id);
+    if *token_manager_account.key != token_manager_check {
+        msg!("Failed to validate token manager account");
+        return Err(ProgramError::InvalidAccountData);
+    }
```

Comment:
Since we already have the `TokenManager` account, we can use the `create_token_manager_pda` instead of the `find_token_manager_pda`. In fact, we already have an `assert_valid_token_manager_pda` function.

```suggestion
    let token_manager = TokenManager::load(token_manager_account)?;
    assert_valid_token_manager_pda(
        token_manager_pda_account,
        &its_root_pda,
        &token_id,
        token_manager.bump,
    )?;
  
```